src/lib.rs

```diff
@@ -4,6 +4,9 @@
 mod base;
 mod error;
 
+#[macro_use]
```

Comment:
As this crate is already using the Rust 2018 edition, could you please import the `new_encoding` directly in src/base.rs via `data_encoding_macro::new_encoding` instead of doing a `#[macro_use]` and `extern crate` here?

Reply:
Sure. Will fix.